### PR TITLE
[fix](Nereids): clone the producer plan and put logicalAnchor generated by `Or_Expansion` above `logicalSink`

### PR DESCRIPTION
**Pull request by keanji-x**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/OrExpansion.java [hunk 1]
```diff
@@ -73,63 +80,109 @@ public class OrExpansion extends OneExplorationRuleFactory {
...
-                .when(join -> supportJoinType.contains(join.getJoinType())
-                        && ConnectContext.get().getSessionVariable().getEnablePipelineEngine())
-                .thenApply(ctx -> {
-                    LogicalJoin<? extends Plan, ? extends Plan> join = ctx.root;
-                    Preconditions.checkArgument(join.getHashJoinConjuncts().isEmpty(),
-                            "Only Expansion nest loop join without hashCond");
+    public Plan rewriteRoot(Plan plan, JobContext jobContext) {
+        OrExpandsionContext ctx = new OrExpandsionContext(
+                jobContext.getCascadesContext().getStatementContext(), jobContext.getCascadesContext());
+        if (!(plan instanceof LogicalResultSink)) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/OrExpansion.java:86]: if we have cte in sql, the root may not be LogicalResultSink. the plan could be
```
CteAnchor1
+-- CteProducer1
+-- CteAnchor2
    +-- CteProducer2
    +-- ResultSink
        +-- ...
```

Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/OrExpansion.java:86]: BTW, u should use `LogicalSink` to support insert into

Comment (XieJiann) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/OrExpansion.java]: fixed, add CTEAnchor above Sink

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

**Review by xzj7019 — APPROVED** [review]

**Comment by XieJiann** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Pull request merged**